string-manipulation-3/src/title-case.js

```diff
@@ -0,0 +1,45 @@
+/* exported titleCase */
+
+function titleCase(string) {
```

Comment:
This is a tough complex problem to solve, great job getting it to work. One thing you can do to help clean it up a bit is to make helper functions for some of the tasks you repeat in multiple places. For example making a function that capitalizes a single word, and I'm sure there are other useful helper function you could make.